tensorflow_graphics/projects/pointnet/train.py

```diff
@@ -130,28 +132,38 @@ def evaluate():
 # ------------------------------------------------------------------------------
 # ------------------------------------------------------------------------------
 
-if not FLAGS.dryrun:
-  ds_train, info = ModelNet40.load(split="train", with_info=True)
+def load_datasets():
+  """Load train / test datasets."""
+  train_dataset, info = ModelNet40.load(split="train", with_info=True)
   num_examples = info.splits["train"].num_examples
-  ds_train = ds_train.shuffle(num_examples, reshuffle_each_iteration=True)
-  ds_train = ds_train.repeat(FLAGS.num_epochs)
-  ds_train = ds_train.batch(FLAGS.batch_size)
-  ds_test = ModelNet40.load(split="test").batch(FLAGS.batch_size)
+  train_dataset = train_dataset.shuffle(
+      num_examples, reshuffle_each_iteration=True)
+  train_dataset = train_dataset.repeat(FLAGS.num_epochs)
+  train_dataset = train_dataset.batch(FLAGS.batch_size)
+  test_dataset = ModelNet40.load(split="test").batch(FLAGS.batch_size)
+  return train_dataset, test_dataset
+
+
+if FLAGS.dryrun:
```

Comment:
Great work! See my comments in the github issue. We *should* be able to remove the dryrun flag system, and have mock_data logic only be in the test file.